tuning/trainingcontroller/controllermetrics/metrics.py

```diff
@@ -0,0 +1,218 @@
+import numpy as np
+from transformers import TrainerState, IntervalStrategy
+from transformers.utils import logging
+from collections import deque
+import math
+import copy
+
+logger = logging.get_logger(__name__)
+
+class MetricHandler:
+    """Base class for the controller-metrics"""
```

Comment:
Don't we want a init to store the name of the handler?

Reply:
This is not required anymore.

Reply:
Added back the name and made related changes to base classes as discussed.